webapp/src/components/rhs/rhs.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState, useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import styled from 'styled-components';
+
+import {makeGetPostsInChannel, getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getPosts} from 'mattermost-redux/actions/posts';
+
+import {manifest} from '@/manifest';
+
+import ThreadItem from './thread_item';
+import RHSHeader from './rhs_header';
+import RHSNewTab from './rhs_new_tab';
+
+const ThreadViewer = styled((window as any).Components.ThreadViewer)`
+    height: 100%;
+`;
+
+const ThreadsList = styled.div`
+    overflow-y: scroll;
+`;
+
+const RhsContainer = styled.div`
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+`;
+
+export default function RHS() {
+    const dispatch = useDispatch();
+    const [currentTab, setCurrentTab] = useState('new');
+
+    const botChannelId = useSelector((state: any) => state['plugins-' + manifest.id].botChannelId);
+    const getPostsInChannel = makeGetPostsInChannel();
+    let posts = useSelector((state) => getPostsInChannel(state as any, botChannelId || '', -1)) || [];
+    posts = posts.sort((a, b) => b.update_at - a.update_at);
+
+    const selectedPostId = useSelector((state: any) => state['plugins-' + manifest.id].selectedPostId);
+
+    useEffect(() => {
+        if (currentTab === 'threads') {
+            dispatch(getPosts(botChannelId || '', 0, 60, false, true, true) as any);
```

Comment:
If CRT is not enabled this fails to get the root posts. 
This needs paging. 
If the bot channel ID is not available we should probably skip instead of fetching with an empty string. Also botChannelId is missing from the useEffect dependency list

Reply:
The lack of paging here, for now, it is intentional to not complicate this too much. It will get the last 60 conversations.

Reply:
Looks like it if it is not enable it gets root and not-root posts, what is not great, but it works, I just need to filter out not-root posts, and also it reduces the amount of posts that we are going to be able to get upfront without paging.

Reply:
I checked and now it works for both, CRT enabled and CRT disabled, but in CRT disabled the list of threads is going to be less. We can add pagination later, or the dedicated API approach that you proposed in another comment.